src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs

```diff
@@ -1439,13 +1439,15 @@ private async Task ReconnectAsync(Exception? closeException)
                         CompleteClose(GetOperationCanceledException("Connection stopped during reconnect attempt. Done reconnecting.", ex, _state.StopCts.Token));
                         return;
                     }
+
+                    previousReconnectAttempts++;
                 }
                 finally
                 {
                     _state.ReleaseConnectionLock();
                 }
 
-                nextRetryDelay = GetNextRetryDelay(previousReconnectAttempts++, DateTime.UtcNow - reconnectStartTime, retryReason);
+                nextRetryDelay = GetNextRetryDelay(previousReconnectAttempts + 1, DateTime.UtcNow - reconnectStartTime, retryReason);
```

Comment:
Do you need the `+ 1` here given we're incrementing earlier in the loop? It looks like the first call to GetNextRetryDelay would be with 0 and the second with 2.

Reply:
Haha, I think this might have been leftover from an intermediate change. If a test catches it good, otherwise I'll add one 😉